backend/siarnaq/api/episodes/models.py

```diff
@@ -270,12 +284,83 @@ class Tournament(models.Model):
     def __str__(self):
         return self.name_short
 
-    def seed_by_scrimmage(self):
+    def get_potential_participants(self):
+        """Returns the list of participants that would be entered in this tournament,
+        if it were to start right now."""
+        # TODO incorporate real eligibility stuff
+        return (
+            Team.objects.with_active_submission()
+            .filter(episode=self.episode)
+            .all()
+            .order_by("-profile__rating__value")
+        )
+
+    def initialize(self):
         """
-        Seed the tournament with eligible teamsn in order of decreasing rating, and
-        populate the Challonge brackets.
+        Seed the tournament with eligible teams in order of decreasing rating,
+        populate the Challonge brackets, and create TournamentRounds.
         """
-        raise NotImplementedError
+
+        tournament_name_public = f"{self.episode.name_long} {self.name_long}"
+        tournament_name_private = tournament_name_public + " (private)"
+
+        # For security by obfuscation,
+        # and to allow easy regeneration of bracket
+        key = random.randint(1000, 9999)
+        # Challonge does not allow hyphens in its IDs
+        # so substitute them just in case
+        tournament_id_public = f"{self.name_short}_{key}".replace("-", "_")
+        tournament_id_private = f"{tournament_id_public}_private"
+
+        # TODO support double
+        is_single_elim = True
+        participants = self.get_potential_participants()
+        # Parse into a format Challonge enjoys
```

Comment:
I kinda don't like that this model has to deal with Challonge things, outside the `challonge` module. It feels a bit coupled to me.

For example, if future devs move away from Challonge, they should be able to swap the `challonge` module out without worrying about this file.

Can this slab of code be moved into the `challonge` module? Perhaps the `bulk_add_participants` method can take a queryset as a function arg.

Reply:
yup yup plan on it -- there was a "# TODO move this block to challonge.py" comment elsewhere, and I meant to put it everywhere, but ya it gonna be future cleanup

Reply:
filed in #549